homeassistant/helpers/config_validation.py

```diff
@@ -376,17 +376,59 @@ def validator(value):
         """Validate ordered dict."""
         config = OrderedDict()
 
-        for key, val in value.items():
-            v_res = item_validator({key: val})
-            config.update(v_res)
+        try:
+            for key, val in value.items():
+                v_res = item_validator({key: val})
+                config.update(v_res)
+        except AttributeError:
+            raise vol.Invalid('Value {} is not a dictionary'.format(value))
 
         return config
 
     return validator
 
 
+def list_or_comma_separated(member_validator=match_all) -> Callable:
+    """Return a validator.
+
+    The validator checks that the value is a list or a comma-separated string
+    that fits member_validator.
+    """
+    def validate(value: Union[str, Sequence]) -> Sequence:
+        """Validate string or sequence."""
+        if value is None:
+            raise vol.Invalid('Value can not be None')
+        if isinstance(value, str):
+            value = [member.strip() for member in value.split(',')]
+        try:
+            return [member_validator(member) for member in value]
+        except (ValueError, TypeError):
+            raise vol.Invalid(
+                'Expected sequence or string, got {}'.format(value))
+
+    return validate
+
+
+def validate_some_keys(keys: Sequence[str], validator: Callable) -> Callable:
+    """Return a validator that validates specified dictionary keys."""
+    def validate(value: Dict) -> Dict:
+        """Validate some of the dictionary keys."""
+        if not isinstance(value, dict):
+            raise vol.Invalid('validate_some_keys require a dict')
```

Comment:
You should not include the name of the validator, only "expects a dict". The voluptuous errors adds it's own context